pages/index.js

```diff
@@ -1,30 +1,35 @@
-import { useState, useEffect } from 'react'
-import FrontPageLayout from '@/components/layouts/FrontPageLayout';
+import { useState, useEffect } from "react";
+import FrontPageLayout from "@/components/layouts/FrontPageLayout";
 import { withIronSessionSsr } from "iron-session/next";
-import { sessionOptions } from '@/lib/Session';
+import { sessionOptions } from "@/lib/Session";
+import { slsFetch } from '@/components/Util'; 
 
-export default function Index() {
-  return (
-    <FrontPageLayout></FrontPageLayout>
-  );
-  }
+export default function Index(props) {
+  return <FrontPageLayout {...props} ></FrontPageLayout>;
+}
 
 export const getServerSideProps = withIronSessionSsr(
   async function getServerSideProps({ req, res }) {
-  
-  let homepage;
-  if(homepage = req.session?.cache?.homepage) {
-    return {
-       redirect: {
-        permanent: false,
-        destination: `/${homepage}`,
-      }     
-    }
-  }
+   try{
+    const response = await slsFetch(`http://localhost:3001/api/resources/mainlogo`);
+    const { data } = await response.json()   
 
-  // Pass data to the page via props
-  return { props: {} }
+    // Pass data to the page via props
+    return { props: { logo: data ?? {} } };
+   }catch(err){
+    return { props: {} }
+   }
+   finally {
+    let homepage;
+    if ((homepage = req.session?.cache?.homepage)) {
+      return {
+        redirect: {
+          permanent: false,
+          destination: `/${homepage}`,
+        },
+      };
+    }
+   }
```

Comment:
I think the `if` checkign is misplaced. It should not be in `finally`. It should be separate from the try-catch of the block where we try to get a logo.
```suggestion
    // declare `data` outside the try-catch block so that it will be accessible outside the block. Set it to null, so that we can do an `if(data)` or `if(logo)` somewhere inside the componentt.
   }catch(err){
     // there's reallly nothing to do if there is an error if you initialize
     // data to null
   }

    let homepage;
    if ((homepage = req.session?.cache?.homepage)) {
      return {
        props:{logo: data }, //probably handle `data is null?` condition inside component.
        redirect: {
          permanent: false,
          destination: `/${homepage}`,
        },
      };
    }
   }
```